### PR TITLE
New: Enable compat-api hints by default

### LINKED ISSUE #1574: [hint-compat-api] Test against live sites
Before turning the new CSS and HTML versions of this hint on-by-default we should do some runs against live sites tracking what it reports to see if there are any obvious false-positives being flagged.

See PR #1596 for a pre-built browser extension bundle for testing.

### PR DESCRIPTION
**Pull request by antross**
## Pull request checklist

Make sure you:

- [x] Signed the [Contributor License Agreement](https://cla.js.foundation/webhintio/hint)
- [x] Followed the [commit message guidelines](https://webhint.io/docs/contributor-guide/getting-started/pull-requests/#commit-messages)

For non-trivial changes, please make sure you also:

- [x] Added/Updated related documentation.
- ~Added/Updated related tests~

## Short description of the change(s)

Fix #1574

Enable `hint-compat-api` by default in
* ~`configuration-development`~
* ~`configuration-web-recommended`~
* `extension-browser`

Bundle for testing:
* [extension-browser-pr-1596.zip](https://github.com/webhintio/hint/files/2847205/extension-browser-pr-1596.zip) (Updated 02/08/2019)

### CODE REVIEW

**Comment by antross** [comment]
@webhintio/core If you review this PR, please also take time to selfhost the included pre-built bundle on some live sites. Specifically I'm interested in identifying any other CSS or HTML features we may want to add to the built-in ignore list before this goes live.

**Comment by molant** [comment]
@antross we should probably add an MDN icon into the HTML formatter same way we do when we use a service or tool from third parties. What do you think?

**Comment by molant** [comment]
I've run the extension in this page with the defaults and I'm getting a bunch of errors :(

```
Uncaught (in promise) DOMException: Failed to execute 'querySelectorAll' on 'Document': 'meta[property="og:description"][content="Pull request checklist
Make sure you:

 Signed the Contributor License Agreement
 Followed the commit message guidelines

For non-trivial changes, please make sure you also:

 Added/Updated related..."]' is not a valid selector.
    at AsyncHTMLDocument.querySelectorAll (file:///C:/Users/molant/Downloads/extension-browser-pr-1596/content-script/webhint.js:61467:51)
    at Object.exports.findElementLocation (file:///C:/Users/molant/Downloads/extension-browser-pr-1596/content-script/webhint.js:33387:50)
    at async Object.exports.findProblemLocation (file:///C:/Users/molant/Downloads/extension-browser-pr-1596/content-script/webhint.js:33446:29)
    at async CompatHTML.<anonymous> (file:///C:/Users/molant/Downloads/extension-browser-pr-1596/content-script/webhint.js:63373:30)
    at async file:///C:/Users/molant/Downloads/extension-browser-pr-1596/content-script/webhint.js:4832:40
```

This is probably not related to the extension itself.

**Comment by antross** [comment]
> I've run the extension in this page with the defaults and I'm getting a bunch of errors :(
> ...
> This is probably not related to the extension itself.

Pretty sure that's a selector generated internally by `hint` when running `findElementLocation` for an in-page element. Looks like we need to escape `\n` during generation (we currently only escape quotes).

**Comment by antross** [comment]
@webhintio/core This is ready for review.

Based on discussions I've scoped the changes to `extension-browser` for now. We'll enable this in `configuration-development` and `configuration-web-recommended` after we've had a chance to get more selfhost feedback through the extension.

File: packages/extension-browser/package.json [hunk 1]
```diff
@@ -9,7 +9,8 @@
   "description": "webhint browser extension",
   "devDependencies": {
     "@hint/hint-axe": "^2.1.1",
-    "@hint/hint-button-type": "^1.0.1",
+    "@hint/hint-button-type": "^2.0.0",
```
Comment (molant) [inline packages/extension-browser/package.json]: Wasn't this version updated when we released yesterday? 🤦‍♂️

**Comment by molant** [comment]
@antross looks like there are conflicts after merging another of your PRs :(

**Comment by antross** [comment]
I'm pretty sure your PR #1884 was the culprit as it renamed the `.md` files I edited here. 😛

**Comment by molant** [comment]
There were conflicts before I merged my PR in `packages/extension-browser/package.json` so I'm not the only one to blame 🙄

**Comment by antross** [comment]
@molant I rebased this, but I want to make sure CI passes before we merge given some of the changes I needed to make.

**Pull request merged**